File: src/main/java/raven/api/hooks/ClientHandler.java
package raven.api.hooks;

import cpw.mods.fml.common.FMLCommonHandler;
import cpw.mods.fml.common.gameevent.TickEvent;
import cpw.mods.fml.relauncher.Side;
import cpw.mods.fml.relauncher.SideOnly;
import net.minecraft.client.multiplayer.WorldClient;
import net.minecraftforge.common.MinecraftForge;
import raven.api.events.ExitEvent;

/**
 * Created by Raven6101 on 19.03.2016.
 */
@SideOnly(Side.CLIENT)
public class ClientHandler {

    public static boolean onPreClientTick(WorldClient worldClient) {
        return FMLCommonHandler.instance().bus().post(new TickEvent.WorldTickEvent(Side.CLIENT, TickEvent.Phase.START, worldClient));
    }

    public static boolean onPostClientTick(WorldClient worldClient) {
        return FMLCommonHandler.instance().bus().post(new TickEvent.WorldTickEvent(Side.CLIENT, TickEvent.Phase.END, worldClient));
    }

    public static boolean onExit() {
        return MinecraftForge.EVENT_BUS.post(new ExitEvent());
    }
}

File: src/main/kotlin/raven/api/client/handler/ClientHandler.kt
package raven.api.client.handler

import cpw.mods.fml.common.FMLCommonHandler
import cpw.mods.fml.common.gameevent.TickEvent
import cpw.mods.fml.relauncher.Side
import net.minecraft.client.multiplayer.WorldClient
import net.minecraftforge.common.MinecraftForge
import raven.api.client.events.ExitEvent

/**
 * Created by r4v3n6101 on 19.03.2016.
 */
object ClientHandler {

    fun onPreClientTick(worldClient: WorldClient): Boolean {
        return FMLCommonHandler.instance().bus().post(TickEvent.WorldTickEvent(Side.CLIENT, TickEvent.Phase.START, worldClient))
    }

    fun onPostClientTick(worldClient: WorldClient): Boolean {
        return FMLCommonHandler.instance().bus().post(TickEvent.WorldTickEvent(Side.CLIENT, TickEvent.Phase.END, worldClient))
    }

    fun onExit(): Boolean {
        return MinecraftForge.EVENT_BUS.post(ExitEvent())
    }
}
